### DISCUSSION TITLE
Welcome to VMPK Discussions!

### DISCUSSION THREAD

**Discussion by pedrolcl** [pedrolcl/VMPK#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by israellh** [comment]
Existe forma de que si tengo multiples teclado USB de computadora conectados a cada uno le asigne un canal midi diferente

**Reply by pedrolcl** [reply]
No es imposible, pero requiere cambios profundos en VMPK para implementarlo. No creo que merezca la pena el esfuerzo.

En primer lugar, se debería obtener eventos de teclado de muy bajo nivel, mucho mas que los métodos implementados actualmente. En segundo lugar, no existe un método que sirva para todos los sistemas operativos, así que se necesitarían implementaciones separadas para cada uno.

Por ejemplo, en los siguientes artículos se explica como hacerlo en Windows:
* https://devblogs.microsoft.com/oldnewthing/20160627-00/?p=93755
* https://devblogs.microsoft.com/oldnewthing/20160926-00/?p=94385

En este proyecto para Linux hay otro método:
* https://github.com/kernc/logkeys

**Reply by pedrolcl** [reply]
Una propuesta mas completa para implementar esta petición:

Todas las entradas y salidas MIDI de VMPK, como las salidas de [dmidiplayer](https://github.com/pedrolcl/dmidiplayer), son plugins de Drumstick::RT. Así que para permitir nuevas entradas o salidas lo que hay que hacer es implementar un nuevo plugin. Hay una [plantilla disponible](https://github.com/pedrolcl/drumstick-rt-outoftree) lista para usar. Navegando a la página y pulsando el boton **"Use this template"** se crea un nuevo proyecto basado en la plantilla con la misma estructura y archivos.

Hay un ejemplo de [plugin independiente](https://github.com/pedrolcl/drumstick/discussions/7) de salida que usa el sintetizador FluidLite: [drumstick-rt-fluidlite](https://github.com/pedrolcl/drumstick-rt-fluidlite).

Otra discusión relacionada está en [añadir Jack MIDI](https://github.com/pedrolcl/drumstick/discussions/9) a VMPK, lo cual se puede resolver mediante el mismo procedimiento.
